net-p2p/qbittorrent/qbittorrent-9999.ebuild

```diff
@@ -56,28 +56,22 @@ BDEPEND+="
 
 DOCS=( AUTHORS Changelog {CONTRIBUTING,README}.md )
 
-src_prepare() {
-	MULTIBUILD_VARIANTS=()
-	use gui && MULTIBUILD_VARIANTS+=( gui )
-	use webui && MULTIBUILD_VARIANTS+=( nogui )
-
-	cmake_src_prepare
-}
-
 src_configure() {
+	MULTIBUILD_VARIANTS+=( $(usev gui) $(usev webui nogui) )
```

Comment:
Maybe it's just me, but although doing it on one line is shorter, I find it somehow harder to read. Consider how we put each element of mycmakeargs on its own line as well.

Reply:
I'm not sure what I think about the change itself, but either way it is not "declarative" and I'm not sure why you are calling it that. :P

You are running a program -- `$(usev)` is not declarative, in fact it's such an impure function it makes use of /dev/stdout.

Reply:
`declare` is not particularly common either! It's effectively an alias for local, unless you toggle it back off with `-g`. Using `declare -g` on its own doesn't really make any sense -- the best reason to use it is if you already needed one of declare's other properties, and the only really useful one I can think of here is `declare -gA`, which you are not actually using here.

So why use declare at all? Just stick with:

```bash
MULTIBUILD_VARIANTS+=(
    $(usev gui)
    $(usev webui nogui)
)
```

Works great, is global by default, looks just like how we assign mycmakeargs, doesn't do conditional re-assignment / append.

Reply:
> Yeah, `-a` excessive and `declare -g` is useless.
> I just wanted to show the fact that it allows the variable to be marked as readonly which is a plus in my book.

```bash
readonly VARNAME="value"
```

;)

But the only matches for `readonly` in the gentoo repository are for e.g. `-mod=readonly` (a golang build flag)...